dask/dataframe/methods.py

```diff
@@ -309,6 +309,7 @@ def concat(dfs, axis=0, join='outer', uniform=False):
                         parts.append(data)
                 out[col] = union_categoricals(parts)
             out = out.reindex(columns=cat_mask.index)
+            ind = concat([df.index for df in dfs2])
```

Comment:
Any way this logic could be merged with the `ind` logic above on lines 260-269?

Reply:
Almost surely, I was being lazy :)

I don't really understand what's going on there. Just unconditionally doing `ind = concat([df.index for df in dfs2])` passes the test suite. I'll spend a bit more time on it.

Reply:
It might not make sense to move it up there, not sure. From the comment it looks like I was attempting to keep all index-related ops there, just thought it might make sense to keep that consistent. IIRC this is because pandas concat works differently for indices than other types.